### ISSUE TITLE
rip_max_cache is uninitialised, resulting in junk in environment

### ISSUE THREAD

**Issue by atsampson** [OpenPrinting/cups#323]
In both `scheduler/cupsfilter.c` and `scheduler/job.c`, there's a `rip_max_cache` variable which gets included in the environment array for subprocesses. However, it's never initialised, so what actually ends up in the environment is some random junk from the stack.

This causes a test failure for me with current master (on amd64 Linux with GCC 11.2), because `error.log` ends up containing binary junk and the test suite's calls to grep report that it's a binary file.

`error.log`:
```
D [12/Jan/2022:11:50:51.840244 +0000] [Job 66] envp[23]="CHARSET=utf-8"
D [12/Jan/2022:11:50:51.840245 +0000] [Job 66] envp[24]="LANG=en.UTF-8"
D [12/Jan/2022:11:50:51.840246 +0000] [Job 66] envp[25]="PPD=/tmp/cups-garstow/ppd/Test1.ppd"
D [12/Jan/2022:11:50:51.840247 +0000] [Job 66] envp[26]="\006ïÔL"
D [12/Jan/2022:11:50:51.840248 +0000] [Job 66] envp[27]="CONTENT_TYPE=image/jpeg"
D [12/Jan/2022:11:50:51.840249 +0000] [Job 66] envp[28]="DEVICE_URI=file:/tmp/Test1"
```

Test suite output:
```
Test Summary

PASS: cupsd exited with no errors.
PASS: All job control files purged.
PASS: Printer 'Test1' correctly produced 131 page(s).
PASS: Printer 'Test2' correctly produced 23 page(s).
PASS: Printer 'Test3' correctly produced 2 page(s).
PASS: 176 requests logged.
PASS: CUPS-Get-Default not logged.
PASS: 0 emergency messages.
PASS: 0 alert messages.
PASS: 0 critical messages.
PASS: 33 error messages.
grep: /tmp/cups-garstow/log/error_log: binary file matches
FAIL: 4 warning messages, expected 14.
W [12/Jan/2022:11:46:13.686312 +0000] No JobPrivateAccess defined in policy default - using defaults.
W [12/Jan/2022:11:46:13.686315 +0000] No JobPrivateValues defined in policy default - using defaults.
W [12/Jan/2022:11:46:13.686316 +0000] No SubscriptionPrivateAccess defined in policy default - using defaults.
W [12/Jan/2022:11:46:13.686317 +0000] No SubscriptionPrivateValues defined in policy default - using defaults.
grep: /tmp/cups-garstow/log/error_log: binary file matches
grep: /tmp/cups-garstow/log/error_log: binary file matches
PASS: 0 notice messages.
grep: /tmp/cups-garstow/log/error_log: binary file matches
PASS: 137 info messages.
grep: /tmp/cups-garstow/log/error_log: binary file matches
PASS: 4243 debug messages.
grep: /tmp/cups-garstow/log/error_log: binary file matches
PASS: 6135 debug2 messages.
grep: /tmp/cups-garstow/log/error_log: binary file matches
```

Removing the `rip_max_cache` code entirely from those two files fixes this for me, but I'm not sure if that's the right fix or if it's intended to contain something...

**Comment by atsampson** [comment]
It looks like ee02b74 only changes `job.c` - `cupsfilter.c` has the same problem...

**Comment by michaelrsweet** [comment]
@atsampson Good catch!

[master 3a1a81d7b] Also fix cupsfilter (Issue #323)

**Comment by atsampson** [comment]
I can confirm that fixes the test suite failure for me - thanks very much!
